### PR TITLE
ci: Fix relayer packaging

### PR DESCRIPTION
**Pull request by palango**
Try to fix the current CI problems

### CODE REVIEW

File: .github/workflows/agent-release.yml [hunk 1]
```diff
@@ -4,6 +4,8 @@ on:
   push:
     tags:
       - '*'
+  pull_request:
```
Comment (istankovic) [inline .github/workflows/agent-release.yml:7]: This is going to publish a Python package on every push, no?

Comment (palango) [reply .github/workflows/agent-release.yml:7]: Yes, this is just for testing.

**Review by istankovic — APPROVED** [review]

File: .github/workflows/agent-release.yml [hunk 2]
```diff
@@ -12,11 +15,16 @@ jobs:
   package:
     runs-on: ubuntu-latest
     timeout-minutes: 10
+    needs: prepare-package
```
Comment (istankovic) [inline .github/workflows/agent-release.yml:18]: Ah, that was the needed incantation... :+1:

File: .github/workflows/prepare-package.yml [hunk 3]
```diff
@@ -21,6 +22,6 @@ jobs:
 
       - uses: actions/upload-artifact@v3
         with:
-          name: source-with-relayers
-          path: .
-          retention-days: 1
+          name: relayers
+          path: relayer/relayer-node16-*
```
Comment (istankovic) [inline .github/workflows/prepare-package.yml:26]: Oh yes, that's exactly what I had in mind, should be plenty quick now!

File: .github/workflows/prepare-package.yml [hunk 4]
```diff
@@ -21,6 +22,6 @@ jobs:
...
+          retention-days: 7
```
Comment (istankovic) [inline .github/workflows/prepare-package.yml:27]: Any reason we like to keep the relayers for 7 days?

File: docker/Dockerfile.agent [hunk 5]
```diff
@@ -2,7 +2,7 @@ FROM python:3.9
 
 COPY . /src
 RUN mkdir -p /src/beamer/data/relayers
-COPY ./relayer/relayer-node16-*-x64 /src/beamer/data/relayers/
+COPY ./relayer-node16-*-x64 /src/beamer/data/relayers/
```
Comment (istankovic) [inline docker/Dockerfile.agent:5]: How can this work? Does local invocation of `make container-image` still work?

Comment (istankovic) [reply docker/Dockerfile.agent:5]: > Why would we ever run this locally?

I have been using this to test our Docker images. Far easier than to mess with github CI.
Also, it is not just about us -- this is intended to be usable by other developers too, and they may well expect this kind of thing to work. Think of the way that we are using Optimism's Docker infrastructure.
Anyone should be able to generate their own, possibly customised, Docker images. Why should we restrict this only to our CI?

Comment (palango) [reply docker/Dockerfile.agent]: Ok, tested it again. works both locally and on ci.

**Pull request merged**